### ISSUE TITLE
Training with SparseBathNorm not giving any result

### ISSUE THREAD

**Issue by HarshitSheoran** [keyu-tian/SparK#93]
I am trying to mod efficientnet-v2s (from timm) with sparse layers

I modded all convolutionals and they work as expected with a similar score to dense
I modded SqueezeExcite block with GlobalAveragePooling and then removing Sparsity from the conv in the SEBlock
I modded the BatchNormAct2D to use an actual layer for BN instead of nn.functional.batch_norm, and this layer is then replaced by SparseBatchNorm2D

The output of original BatchNormAct2D and the new class with SparseBatchNorm2D inplace is the same but while training SparseBatchNorm2D doesn't learn

`
from timm.layers.trace_utils import _assert
class BatchNormAct2dMod(nn.BatchNorm2d):
    """BatchNorm + Activation

    This module performs BatchNorm + Activation in a manner that will remain backwards
    compatible with weights trained with separate bn, act. This is why we inherit from BN
    instead of composing it as a .bn member.
    """
    def __init__(
            self,
            num_features,
            eps=1e-5,
            momentum=0.1,
            affine=True,
            track_running_stats=True,
            apply_act=True,
            act_layer=nn.ReLU,
            act_kwargs=None,
            inplace=True,
            drop_layer=None,
            device=None,
            dtype=None,
    ):
        try:
            factory_kwargs = {'device': device, 'dtype': dtype}
            super(BatchNormAct2dMod, self).__init__(
                num_features,
                eps=eps,
                momentum=momentum,
                affine=affine,
                track_running_stats=track_running_stats,
                **factory_kwargs,
            )
        except TypeError:
            # NOTE for backwards compat with old PyTorch w/o factory device/dtype support
            super(BatchNormAct2dMod, self).__init__(
                num_features,
                eps=eps,
                momentum=momentum,
                affine=affine,
                track_running_stats=track_running_stats,
            )
        
        self.bn = nn.BatchNorm2d(num_features=num_features, eps=eps, momentum=momentum, affine=affine, track_running_stats=track_running_stats)
        #self.bn = SparseBatchNorm2d(num_features=num_features, eps=eps, momentum=momentum, affine=affine, track_running_stats=track_running_stats)
        self.drop = drop_layer() if drop_layer is not None else nn.Identity()
        self.act = _create_act(act_layer, act_kwargs=act_kwargs, inplace=inplace, apply_act=apply_act)

    def forward(self, x):
        #print(f"inp x {x[0][0][160]}")
        x = self.bn(x)
        #print(f"bn x {x[0][0][160]}")
        x = self.drop(x)
        #print(f"drop x {x[0][0][160]}")
        x = self.act(x)
        #print(f"act x {x[0][0][160]}")
        return x
`

Is there anything I am doing wrong or is there another piece of code that shows how sparsity with Efficientnetv2-S can be achieved?

Thank you
